plugins/hydra_ax_sweeper/hydra_plugins/hydra_ax_sweeper/ax_sweeper.py

```diff
@@ -0,0 +1,303 @@
+            OmegaConf.save(OmegaConf.create(results_to_serialize), f)
+        log.info("Best parameters: " + str(best))
+
+    def sweep_over_batches(
+        self,
+        ax_client: AxClient,
+        overrides: Sequence[Sequence[str]],
+        batch_of_trials_to_launch: BatchOfTrialType,
+    ) -> None:
+        rets = self.launcher.launch(overrides)  # type: ignore
+        for idx in range(len(batch_of_trials_to_launch)):
+            val = rets[idx].return_value
+            ax_client.complete_trial(
+                trial_index=batch_of_trials_to_launch[idx].trial_index, raw_data=val,
+            )
+
+    def setup_ax_client(self, arguments: List[str]) -> AxClient:
+        """Method to setup the Ax Client"""
+
+        parameters: List[Dict[str, Any]] = []
+        for key, value in self.ax_params.items():
+            param = OmegaConf.to_container(value, resolve=True)
+            assert isinstance(param, Dict)
+            if param["type"] == "range":
+                bounds = param["bounds"]
+                if not (all(isinstance(x, int) for x in bounds)):
+                    # Type mismatch. Promote all to float
+                    param["bounds"] = [float(x) for x in bounds]
+
+            parameters.append(param)
+            parameters[-1]["name"] = key
+        commandline_params = self.parse_commandline_args(arguments)
+        for cmd_param in commandline_params:
+            for param in parameters:
+                if param["name"] == cmd_param["name"]:
+                    for key, value in cmd_param.items():
+                        param[key] = value
+        ax_client = AxClient(
+            verbose_logging=self.verbose_logging, random_seed=self.random_seed
+        )
+        ax_client.create_experiment(parameters=parameters, **self.experiment)
+
+        return ax_client
+
+    def parse_commandline_args(
+        self, arguments: List[str]
+    ) -> List[Dict[str, Union[ax_types.TParamValue, List[ax_types.TParamValue]]]]:
+        """Method to parse the command line arguments and convert them into Ax parameters"""
+
+        parameters = []
+        for arg in arguments:
+            key, value = arg.split("=")
+            if "," in value:
+                # This is a Choice Parameter.
+                value_choices = [x.strip() for x in value.split(",")]
+                if all(_is_int(x) for x in value_choices):
+                    param = {
+                        "name": key,
+                        "type": "choice",
+                        "values": [int(x) for x in value_choices],
+                        "parameter_type": ParameterType.INT,
+                    }
+                elif all(_is_float(x) for x in value_choices):
+                    param = {
+                        "name": key,
+                        "type": "choice",
+                        "values": [float(x) for x in value_choices],
+                        "parameter_type": ParameterType.FLOAT,
+                    }
```

Comment:
Looking here, it appears that the command line decides on parameter_type.
I suggest something else:

If parameter_type is specified in the config, this is it. the command line should be converted to it.
if it's not specified, you infer it from the types.

Don't forget to add tests covering this logic.


This is a bit much. we can do it in a followup diff after we land this one.

Reply:
No. cmd is supposed to override the config. Why would we want to confuse the user by requiring them to specify the type via the config file? Doesnt that break the entire point of using cmd?